homeassistant/components/zha/core/gateway.py

```diff
@@ -393,7 +393,10 @@ def device_removed(self, device: zigpy.device.Device) -> None:
             device_info = zha_device.zha_device_info
             zha_device.async_cleanup_handles()
             async_dispatcher_send(self._hass, f"{SIGNAL_REMOVE}_{str(zha_device.ieee)}")
-            asyncio.ensure_future(self._async_remove_device(zha_device, entity_refs))
+            self._hass.async_create_task(
```

Comment:
Is `self._hass.async_create_task` the preferred method? Or could we instead tie this to the config entry with `self.config_entry.async_create_background_task(self._hass, self._async_remove_device(zha_device, entity_refs), "ZHAGateway._async_remove_device")`?

Reply:
I chose create_task on purpose because background tasks get cancelled on config entry unload/HA shutdown. I wouldn't want to cancel the removal of a device.